### DISCUSSION TITLE
Can we set priority to the POI label (point of interest)?  To display labels & icon for selected or rendered POI's?

### DISCUSSION THREAD

**Discussion by satya-auti** [maplibre/maplibre-gl-js#2344] [Q&A]
Can we set priority to the POI label (point of interest)?  To display labels for selected or rendered POI's?

I have show the stable POI label on map. Once the poi is display on the map after a specific zoom then it needs to be stable till the last zoom (max zoom).

Is this possible using maplibre-gl-js?

**Comment by HarelM** [comment]
You can draw a label/poi on top of the existing ones I think. Not sure I fully understand the question here.

**Reply by satya-auti** [reply]
Actually I have to show the poi with label on map which has higher popularity/most visited value. Once its display on map then it should be stable and display till the max zoom.

**Reply by wipfli** [reply]
https://github.com/apache/incubator-baremaps/issues/585#issuecomment-1431652498

**Reply by wipfli** [reply]
I think the problem is this bubbling effect when you have too many labels close together. Usually people make sure that they do not include too many labels in the tiles.

**Reply by satya-auti** [reply]
Yes same thing happening here,  when zooming in, they appear and disappear again because one label eats the other and gets eaten by a third such that the other appears again..


Is there any solution for poi to stay stable on the map once it is rendered?

**Reply by satya-auti** [reply]
> [apache/incubator-baremaps#585 (comment)](https://github.com/apache/incubator-baremaps/issues/585#issuecomment-1431652498)

Facing same issue with poi. Is there any solution for this?

**Reply by wipfli** [reply]
Yes, you can control point density in Planetiler with something like this:

https://github.com/onthegomap/planetiler/blob/b01975bd43d220dec9cebaf200639ad600aeb798/planetiler-examples/src/main/java/com/onthegomap/planetiler/examples/ToiletsOverlay.java#L43-L50

**Reply by satya-auti** [reply]
> I think the problem is this bubbling effect when you have too many labels close together. Usually people make sure that they do not include too many labels in the tiles.

I also don't want to display the many labels but at least need to display prioritize poi on the map and which should be stable on the map (once appear should not disappear again).

**Reply by satya-auti** [reply]
can we done this without planetiler? (only with maplibre).

**Reply by wipfli** [reply]
No, because this is problem which has to do with the tiles I would say
